### PR TITLE
➡️ Migrate core package 'dalek' into ./packages

### PR DESCRIPTION
**Pull request by daviwil**
### Description of the Change

This PR migrates the core [`dalek`](https://github.com/atom/dalek) package into `atom/atom`.  See issue #17838 for more information.

### CODE REVIEW

**Review by jasonrudolph — APPROVED** [review]

File: packages/dalek/lib/main.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+const Grim = require('grim')
+
+module.exports = {
+  activate () {
+    atom.packages.onDidActivateInitialPackages(async () => {
+      const duplicates = await dalek.enumerate()
+      for (let i = 0; i < duplicates.length; i++) {
+        const duplicate = duplicates[i]
+        Grim.deprecate(
+          `You have the core package "${duplicate}" installed as a community package. See https://github.com/atom/dalek for how this causes problems and instructions on how to correct the situation.`,
```
Comment (Ben3eeE) [inline packages/dalek/lib/main.js]: This link is there to show the users the steps in the readme so they can resolve the issue. Should it be changed to the new location of the readme file?

Comment (daviwil) [reply packages/dalek/lib/main.js]: Yep, thanks for catching that!

Comment (jasonrudolph) [reply packages/dalek/lib/main.js]: Nice catch, @Ben3eeE. :zap:

**Pull request merged**